Sofa/framework/Type/src/sofa/type/RGBAColor.h

```diff
@@ -67,33 +91,56 @@ class SOFA_TYPE_API RGBAColor : public fixed_array<float, 4>
     /// @brief enlight a color by a given factor.
     static RGBAColor lighten(const RGBAColor& in, const SReal factor);
 
-    constexpr float& r(){ return this->elems[0] ; }
-    constexpr float& g(){ return this->elems[1] ; }
-    constexpr float& b(){ return this->elems[2] ; }
-    constexpr float& a(){ return this->elems[3] ; }
-    constexpr const float& r() const { return this->elems[0] ; }
-    constexpr const float& g() const { return this->elems[1] ; }
-    constexpr const float& b() const { return this->elems[2] ; }
-    constexpr const float& a() const { return this->elems[3] ; }
-
-    constexpr void r(const float r){ this->elems[0]=r; }
-    constexpr void g(const float g){ this->elems[1]=g; }
-    constexpr void b(const float b){ this->elems[2]=b; }
-    constexpr void a(const float a){ this->elems[3]=a; }
+    constexpr float& r(){ return this->m_components[0] ; }
+    constexpr float& g(){ return this->m_components[1] ; }
+    constexpr float& b(){ return this->m_components[2] ; }
+    constexpr float& a(){ return this->m_components[3] ; }
+    constexpr const float& r() const { return this->m_components[0] ; }
+    constexpr const float& g() const { return this->m_components[1] ; }
+    constexpr const float& b() const { return this->m_components[2] ; }
+    constexpr const float& a() const { return this->m_components[3] ; }
+
+    constexpr void r(const float r){ this->m_components[0]=r; }
+    constexpr void g(const float g){ this->m_components[1]=g; }
+    constexpr void b(const float b){ this->m_components[2]=b; }
+    constexpr void a(const float a){ this->m_components[3]=a; }
+
+    // operator[]
+    constexpr float& operator[](std::size_t i)
+    {
+#ifndef NDEBUG
+        assert(i < N && "index in RGBAColor must be smaller than 4");
+#endif
+        return m_components[i];
+    }
+    constexpr const float& operator[](std::size_t i) const
+    {
+#ifndef NDEBUG
+        assert(i < N && "index in RGBAColor must be smaller than 4");
+#endif
```

Comment:
```suggestion
        assert(i < N && "index in RGBAColor must be smaller than 4");
```
`assert` is already conditioned to `NDEBUG` no?

Reply:
should be, I just copy-pasted brainlessly